addons/binding/org.openhab.binding.yamahareceiver2/src/main/java/org/openhab/binding/yamahareceiver2/discovery/ZoneDiscoveryService.java

```diff
@@ -0,0 +1,61 @@
+package org.openhab.binding.yamahareceiver2.discovery;
+
+import static org.openhab.binding.yamahareceiver2.YamahaReceiverBindingConstants.THING_TYPE_YAMAHAAV;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.yamahareceiver2.YamahaReceiverBindingConstants;
+import org.openhab.binding.yamahareceiver2.internal.YamahaReceiverState;
+import org.openhab.binding.yamahareceiver2.internal.protocol.YamahaReceiverCommunication.Zone;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+
+public class ZoneDiscoveryService extends AbstractDiscoveryService {
```

Comment:
Other zones do have the same channels as the main zone, and not every receiver has the same zones. I actually only now which zones are available after the first state fetch.
Is it possible to add a channel group dynamically like it is with things? I'm not yet familiar with channel groups but will read the documentation.